store/src/sql/sqlite/event.rs

```diff
@@ -1,38 +1,60 @@
-use std::collections::BTreeSet;
+use std::{
+    collections::BTreeSet,
+    num::TryFromIntError,
+    sync::{atomic::AtomicI64, Arc},
+};
 
-use anyhow::{anyhow, Context, Result};
+use anyhow::anyhow;
 use async_trait::async_trait;
 use bytes::Bytes;
-use ceramic_core::{EventId, RangeOpen};
+use ceramic_core::{event_id::InvalidEventId, EventId, RangeOpen};
 
 use cid::Cid;
 use iroh_bitswap::Block;
 use iroh_car::CarReader;
 use multihash::{Code, Multihash, MultihashDigest};
-use recon::{AssociativeHash, HashCount, InsertResult, Key, ReconItem, Sha256a};
+use recon::{
+    AssociativeHash, Error as ReconError, HashCount, InsertResult, Key, ReconItem,
+    Result as ReconResult, Sha256a,
+};
 
 use tracing::instrument;
 
 use crate::{
     sql::{
-        rebuild_car, BlockBytes, BlockQuery, BlockRow, CountRow, DeliveredEvent, EventIdError,
-        EventQuery, EventValueRaw, FirstAndLast, OrderKey, ReconHash, ReconQuery, ReconType,
-        SqlBackend, GLOBAL_COUNTER,
+        rebuild_car, BlockBytes, BlockQuery, BlockRow, CountRow, DeliveredEvent, EventQuery,
+        EventValueRaw, FirstAndLast, OrderKey, ReconHash, ReconQuery, ReconType, SqlBackend,
+        GLOBAL_COUNTER,
     },
-    DbTxSqlite, SqlitePool,
+    DbTxSqlite, Error, Result, SqlitePool,
 };
 
 /// Unified implementation of [`recon::Store`] and [`iroh_bitswap::Store`] that can expose the
 /// individual blocks from the CAR files directly.
 #[derive(Clone, Debug)]
 pub struct SqliteEventStore {
     pub(crate) pool: SqlitePool,
+    test_counter: Option<Arc<AtomicI64>>,
 }
 
 impl SqliteEventStore {
     /// Create an instance of the store initializing any neccessary tables.
     pub async fn new(pool: SqlitePool) -> Result<Self> {
-        let store = SqliteEventStore { pool };
+        let store = SqliteEventStore {
+            pool,
+            test_counter: None,
+        };
+        store.init_delivered().await?;
+        Ok(store)
+    }
+
+    /// Creates an instance that doesn't share the global event counter with other instances (only for tests to avoid running serially)
+    #[allow(dead_code)]
+    pub(crate) async fn new_local(pool: SqlitePool) -> Result<Self> {
+        let store = SqliteEventStore {
+            pool,
+            test_counter: Some(Arc::new(AtomicI64::new(0))),
```

Comment:
Can we mock this for tests instead, by mocking the delivered function?

Reply:
Yeah, this field and function should be deleted.. it was related to postgres and sharing a single DB with things running in parallel and trying to assert highwater marks. But there are definitely better ways to manage that and it felt pretty janky.

I should probably rebase all of this off of main. #320 doesn't add a ton of short term value and some things like this leaked in 😞.

Reply:
Removed this field. I'm still targeting the other PR but I adjusted it to make it a bit easier to see what was refactored.. github wasn't showing the refactor + move nicely. It's still way too big 🙈